include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1230 @@
+            return false;
+        }
+        elseif (count($results) == 0) { // No events match. Return emtpy array.
+            return Array();
+        }
+        else {
+            $gEvent = $results[0];
+            return $gEvent;
+        }
+    }
+
+    /**
+     * Get a google event by the event id
+     *
+     * @param string $event_id Google Event ID
+     *
+     * @return Google_Service_Calendar_Event|null Google_Service_Calendar_Event if found, null if not found
+     */
+    public function getGoogleEventById($event_id)
+    {
+
+        // Make sure the calendar service is set up
+        if (!isset($this->gService))
+        {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
+            return false;
+        }
+
+        if (empty($event_id))
+        { // If we didn't get passed an event, return null
+            return;
+        }
+
+        $gEvent = $this->gService->events->get($this->calendarId, $event_id);
+
+        if (!empty($gEvent->getId()))
+        {
+            return $gEvent;
+        }
+        else {
+            return;
+        }
+    }
+
+    /**
+     * Find a missing event on users calendars
+     *
+     * This is an expensive method. It searches *all* the users caleners for a missing event,
+     * moves it back to the SuiteCRM calendar, and returns the moved event.
+     *
+     * @param string $event_id The Google Event Id
+     *
+     * @return Google_Service_Calendar_Event
+     */
+    public function getMissingMeeting($event_id)
+    {
+        // get list of users calendars
+        $calendarList = $this->gService->calendarList->listCalendarList();
+
+        // Search all calendars looking for event
+        foreach ($calendarList->getItems() as $calendarListEntry)
+        {
+            if ($gEvent = $this->gService->events->get($calendarListEntry->getId(), $event_id))
+            {
+                $foundOnCalendar = $calendarListEntry->getId();
+                break;
+            }
+        }
+        if (isset($foundOnCalendar))
+        {
+            $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Found Event on Calendar ID: ' . $foundOnCalendar);
+            $result = $this->gService->events->move($foundOnCalendar, $event_id, $this->calendarId);
+            return $result;
+        }
+        else {
+            $GLOBALS['log']->warn(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to find missing Calendar Event');
+            return;
+        }
+    }
+
+    /**
+     * Get a SuiteCRM meeting by Google Event ID
+     *
+     * @param string $event_id The Google Event ID
+     *
+     * @return Meeting
+     */
+    public function getMeetingByEventId($event_id)
+    {
+
+        // We do it this way so we also get deleted meetings
+        $query = "SELECT id FROM meetings WHERE gsync_id = '" . $event_id . "'";
```

Comment:
Please consider using single quotes or alternatively remove the concatenation. 
`"SELECT id FROM meetings WHERE gsync_id = '{$event_id}'"`

Reply:
Removing the concatenation